### PR TITLE
Video Optimization: Display Snackbar Message

### LINKED ISSUE #6066: Video Optimization: Display Snackbar Message
## Feature Description

Display a snackbar notification during video transcoding, saying that the file is being optimized.

> Video optimization in progress.

Screenshot 2021-02-15 at 18 02 08

**[View on Figma](https://www.figma.com/file/NHHMwIfxMnz39NxqkrFb7R/Stories-Post-Stable?node-id=5511%3A441091)**

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance Criteria

* Telemetry: clicks on the snackbar CTA links are being tracked
* If optimization is skipped, just upload the originally provided file instead
* When cancelling, remove the video element from story / media library

## Implementation Brief

**`useTranscodeVideo.js`**

* Use `const [isProcessing, setIsProcessing] = useState(false);`
* Call `setIsProcessing()` appropriately in `transcodeVideo()`
* Return `isProcessing` from the hook

**`useUploadMedia.js`**

* Use `const { isProcessing } = useTranscodeVideo();`
* Use `useEffect` to call `showSnackbar()` if `isProcessing`

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

Display a snackbar notification during video transcoding, saying that the file is being optimized.

## Relevant Technical Choices

I did not add a cross button, as the snackbar message automatically hides after 5 seconds ( is this the current behaviour? ). 

Without larger changes to the snackbar message, I way not able to position the message bottom center. 

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6066

### CODE REVIEW

File: assets/src/edit-story/app/uploader/useUploader.js [hunk 1]
```diff
@@ -162,13 +163,16 @@ function useUploader() {
...
+        setIsProcessing(true);
         // TODO: Only transcode & optimize video if needed (criteria TBD).
         const newFile = await transcodeVideo(file);
         trackTiming();
+        setIsProcessing(false);
         additionalData.media_source = 'video-optimization';
         return uploadMedia(newFile, additionalData);
       } catch (err) {
         trackTiming();
+        setIsProcessing(false);
```
Comment (swissspidy) [inline assets/src/edit-story/app/uploader/useUploader.js]: Just do this in a `finally`?

**Comment by swissspidy** [comment]
> Had to move the isProcessing state to useUploader is in a try / catch, it made sense to put it there, incase the transcoding fails.

I see.

Hmm when I look at that,  I think `isTranscoding`/`setIsTranscoding` is more descriptive.

> I did not add a cross button, as the snackbar message automatically hides after 5 seconds ( is this the current behaviour? ).
> Without larger changes to the snackbar message, I way not able to position the message bottom center.

The snackbars will be updated to use the new ones from the design system, we don't need to worry about that.

File: assets/src/edit-story/app/uploader/useUploader.js (hunk 1 above)
Comment (spacedmonkey) [reply assets/src/edit-story/app/uploader/useUploader.js]: We have the same problem that we had with the tracking. The return on 172.

**Comment by spacedmonkey** [comment]
> Hmm when I look at that, I think isTranscoding/setIsTranscoding is more descriptive.

Using state to push snackbar messages meant that, I ended up pushing hundreds of snack bar messages in a loop. I have pushed a much more simple solution that seems to work well. 

Can you review again?

**Comment by spacedmonkey** [comment]
This was the error I was getting in the first implementation. 

> Error: Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate. React limits the number of nested updates to prevent infinite loops.

File: assets/src/edit-story/app/uploader/useUploader.js (hunk 1 above)
Comment (swissspidy) [reply assets/src/edit-story/app/uploader/useUploader.js]: Dang, I looked for a `return` but I didn't see it 🤦

Comment (swissspidy) [reply assets/src/edit-story/app/uploader/useUploader.js]: Actually, the return is in the `try` block, not in the `catch` block. So doing the update in `finally` is totally possible without getting unexpected results.

Comment (spacedmonkey) [reply assets/src/edit-story/app/uploader/useUploader.js]: I get out a lint error, when I put it in.

**Review by swissspidy — APPROVED** [review]

**Pull request merged**